api/src/main/java/org/openmrs/module/idgen/LocationPrefixedSequentialIdentifierGenerator.java

```diff
@@ -0,0 +1,53 @@
+package org.openmrs.module.idgen;
+
+import org.apache.commons.lang.StringUtils;
+import org.openmrs.api.InvalidIdentifierFormatException;
+import org.openmrs.api.context.Context;
+
+public class LocationPrefixedSequentialIdentifierGenerator extends SequentialIdentifierGenerator {
+
+	private final String DEFAULT_LOCATION_PREFIXED_IDENTIFIER_FORMAT = "[A-Z]{3,5}\\-[0]{3}\\-[0-9]{3,6}";  
+	
+	@Override
+    public String getIdentifierForSeed(long seed) {
```

Comment:
You don't take any profit of inheritance here, why don't you do something like
```java
@Override
public String getIdentifierForSeed(long seed) {
  setPrefix(null);
  ...
  String prefix = ... // get it recursively
  ...
  return prefix + super.getIdentifierForSeed(seed);
}
```
?